### PR TITLE
[observability] Log channel capacity every 5 sec

### PR DESCRIPTION
**Pull request by gdanezis**
Added a periodic (5 sec) log message with the capacity of the primary core channels. In case one of them is at zero, we emit an error as this might lead to deadlock.

### CODE REVIEW

**Review by huitseeker — DISMISSED** [review]
YES!

**Comment by gdanezis** [comment]
Yep feel free to refactor to use something more legit indeed

**Comment by gdanezis** [comment]
Ok, it seems that the approach by which a task keeps hold of channel senders does not play well with some reconfig work -- so some tests do not terminate @asonnino

File: primary/src/primary.rs [hunk 1]
```diff
@@ -103,6 +104,59 @@ impl Primary {
...
+        let mon_tx_get_block_commands = tx_get_block_commands.clone();
+        let mon_tx_batches = tx_batches.clone();
+        let mon_tx_block_removal_commands = tx_block_removal_commands.clone();
+        let mon_tx_batch_removal = tx_batch_removal.clone();
+        let mon_tx_block_synchronizer_commands = tx_block_synchronizer_commands.clone();
+        let mon_tx_certificate_responses = tx_certificate_responses.clone();
+        let mon_tx_payload_availability_responses = tx_payload_availability_responses.clone();
+        let mon_tx_state_handler = tx_state_handler.clone();
+
+        tokio::task::spawn(async move {
```
Comment (huitseeker) [inline primary/src/primary.rs:127]: This doesn't work well: you want to collect the task created in this line, and put that handle (like all the task handles created in this file) in the `handles` vector at the end of this function.

Comment (gdanezis) [reply primary/src/primary.rs]: And also need too shutdown if told so. New patch incoming.

**Review by huitseeker — APPROVED** [review]
Opened https://github.com/MystenLabs/narwhal/issues/655 for follow-up, thanks @gdanezis !

**Pull request merged**